### PR TITLE
Minimal fix for crash #872

### PR DESCRIPTION
**Pull request by 1fish2**
**Summary:** 

Scrolling down the 2018 and 2019 Districts screen to where "FIRST In Texas District" should appear will crash.

**Issues Reference:** 
#872 

**Test Plan:** 
Manual test.

**Screenshots:**
screenshot_1537073111

### CODE REVIEW

**Comment by 1fish2** [comment]
This PR does not try to update travis-ci nor fully fix the build files. See PR #861 for a broader attempt, if not 100% successful.

**Comment by 1fish2** [comment]
Oops, I thought I pushed the whole fix the other day.

That's the minimal fix. One could go a little further and log an unrecognized district code. Also of course update the travis CI build. I could take a guess at that but it's a mystery to me.

**Comment by sorianog** [comment]
@1fish2 You still planning on going forward with this PR? Looks like Travis ain't happy 😢

**Comment by sorianog** [comment]
@1fish2 Hmm, I can take a look at the tests after you're done.

**Comment by 1fish2** [comment]
@sorianog I pushed a cut at making travis-ci work. Please do look at the tests. Many thanks!

We must push a release before Nov. 1. Starting Nov. 1, releases must support API 26. That's where Robolectric, library versions, and Android changes hit the fan.

**Comment by sorianog** [comment]
@1fish2 Well, I'm glad I'm here to help since what you explained is definitely something I helped work on ("app currency" - keep app updated with latest libs/tools/frameworks) at my full-time gig. Esp, Roboelectric stuff. Want to be careful with the new setup for injections and what not. I'll keep that deadline in mind!

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Fri Sep 16 18:38:07 EDT 2016
+#Sat Sep 15 19:45:22 PDT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (sorianog) [inline gradle/wrapper/gradle-wrapper.properties]: Ah, good this is a def a req for increasing your API level support. Gradle needs to be updated as you did here!

File: libImgur/build.gradle [hunk 2]
```diff
@@ -4,6 +4,6 @@ sourceCompatibility = 1.6
 targetCompatibility = 1.6
 
 dependencies {
-    compile "com.squareup.retrofit2:retrofit:${retrofitVersion}"
-    compile "io.reactivex:rxjava:${rxJavaVersion}"
-}
\ No newline at end of file
+    implementation "com.squareup.retrofit2:retrofit:${retrofitVersion}"
+    implementation "io.reactivex:rxjava:${rxJavaVersion}"
```
Comment (sorianog) [inline libImgur/build.gradle]: I like how vars are used here to dynamically change the version of chosen libs/frameworks. Is this something that can be done with more of the libs in this project?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (1fish2) [reply gradle/wrapper/gradle-wrapper.properties]: Terrific! Welcome!

I have another cut at updating for Robolectric compatibility. I'll push that shortly after trying further to work through the errors that arise at each attempt.

Their [migration guide](http://robolectric.org/migrating/) is somewhat helpful. The good news is they've updated Robolectric for current or almost-current Android APIs. The bad news is various parts of their APIs changed and various other tools and libs have to upgrade to get there.

In general, the Android gradle rules (esp. for this project) and library dependencies are a mystery to me. Docs say we need to turn on `includeAndroidResources = true` but does that mean we should turn off the `build.workaround-missing-resource.gradle`?

File: libImgur/build.gradle (hunk 2 above)
Comment (1fish2) [reply libImgur/build.gradle]: Phil put this in. I surmise that it's particularly useful for the nested projects libImgur and libTba. I don't know why those are nested projects.

The build still gets into trouble when dependent libraries use different versions of these dependencies, not to mention outdated gradle commands like `compile` and task `<<`.

Comment (1fish2) [reply libImgur/build.gradle]: It's particularly useful with `${supportLibVersion}`. It does add complexity and surprises, like when gradle complains that we're not using the same support library version everywhere (which must be due to an indirect dependency), so I'd want to extend it only to places where the same version number must appear in multiple places.

Comment (1fish2) [reply libImgur/build.gradle]: Do the current build tools make retrofit obsolete?

**Comment by 1fish2** [comment]
@sorianog I'm out of ideas to try other than updating library versions, which might have n^2 compatibility problems but it might fix some of the build problems like `Configuration 'compile' is obsolete`. So if you can make progress, please do.

**Comment by sorianog** [comment]
@1fish2 Thanks for the work thus far! I'll see what I can do.

File: libImgur/build.gradle (hunk 2 above)
Comment (sorianog) [reply libImgur/build.gradle]: @1fish2 Nah, I think retrofit is still okay to include. That's how it is at my work's codebase at least!

**Comment by phil-lopreiato** [comment]
I removed the hard-coded districts entirely in https://github.com/the-blue-alliance/the-blue-alliance-android/pull/874 so I'm going to close this, but let's definitely figure out all the inane things we need to do to get the app to build with new tools

**Pull request closed without merging**